ipfs/mock.go

```diff
@@ -26,8 +28,9 @@ func NewMockNode() (*core.IpfsNode, error) {
 
 	// Effectively offline, only peer in its network
 	return core.NewNode(ctx, &core.BuildCfg{
-		Online: true,
-		Host:   MockHostOption(mocknet.New(ctx)),
+		Online:  true,
+		Host:    MockHostOption(mocknet.New(ctx)),
+		Routing: nilrouting.ConstructNilRouting,
```

Comment:
That allows the Go tests to write to IPFS without an error. Otherwise when trying to add the inventory to IPFS it blows up with an error about not having a routing system instantiated. This adds a mock router which lets us write to IPFS but does not allow communicating with other nodes. I think this is a good choice for future IPFS testing as well.

Reply:
Awesome.